### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by janezpodhostnik**
This fix is three fixes in a trench coat:
1. Fix flakiness of compactor tests. The root cause was that the wal still had a lock on the test dir when the test exited, because we explicitly decide not to wait for it in the code. I added a max 10ms wait, which makes the tests 100% stable (up from 75% stable) (tested on 100 runs on my machine)
2. network/test/cohort2/unicast_authorization_test.go the reasoning is in the comments. I'm least sure about this one.
3. connection/cache.go. The problem was that wg.done() was being called after wg was already finished. I added a lock on the closeRequested and wg, which is probably going to be a slight performance hit, since we used to use atomics for the `closeRequested`

## Summary by CodeRabbit

* **Tests**
  * Refactored concurrent access testing with batched approach for improved performance validation.
  * Updated error tolerance in authorization tests to handle edge cases.
  * Improved test resilience with proper public method accessors.

* **Bug Fixes**
  * Enhanced shutdown logic with bounded timeout for checkpoint operations to improve stability.

### CODE REVIEW

File: engine/access/rpc/connection/cache.go [hunk 1]
```diff
@@ -7,7 +7,6 @@ import (
 	lru "github.com/hashicorp/golang-lru/v2"
```
Comment (peterargue) [inline engine/access/rpc/connection/cache.go:7]: I'm attempting to fix this in another PR by completely refactoring the connection cache:
https://github.com/onflow/flow-go/pull/7859

it needs some more rounds of review though.

my preference is to leave this logic as is, and push that PR forward to fix the tests.

Comment (janezpodhostnik) [reply engine/access/rpc/connection/cache.go:7]: I'll take a look :eyes:. 

Would you be ok with adding a delay in the test to temporarily address the issue?

Are you ok with the other two?

**Review by zhangchiqing — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
@peterargue can you do another quick glance here. I checked and everything is still relevant.

**Review by peterargue — APPROVED** [review]
